### PR TITLE
allow nbclient 0.10

### LINKED ISSUE #1429: Chore Update to use nbclient 0.9.0
## Description

nbclient 0.9.0 was release in 2023-11, current version of voila is pinned to <9.0

## Reproduce

2. Click on '...'
3. Scroll down to '...'
4. See error '...'

## Context

- voila version 
- Operating System and version:
- Browser and version:

<details><summary>Troubleshoot Output</summary>
<pre>
Paste the output from running `jupyter troubleshoot` from the command line here.
You may want to sanitize the paths in the output.
</pre>
</details>

<details><summary>Command Line Output</summary>
<pre>
Paste the output from your command line running `voila` here, use `--debug` if possible.
</pre>
</details>

<details><summary>Browser Output</summary>
<pre>
Paste the output from your browser Javascript console here.
</pre>
</details>

### If using JupyterLab
- JupyterLab version:

<details><summary>Installed Labextensions</summary>
<pre>
Paste the output from your command line running `jupyter labextension list`.
</pre>
</details>

### PR DESCRIPTION
**Pull request by minrk**
## References

closes #1429

## Code changes

increases nbclient upper bound from 0.8 to 0.11.

## User-facing changes

current nbclient (0.10) is accepted

## Backwards-incompatible changes

None

### CODE REVIEW

**Comment by minrk** [comment]
Since tests pass, which means this upper bound was unnecessarily restrictive, I'd recommend removing the upper bound entirely, but I'll leave that to maintainers. I don't believe upper bounds in packages is a practice to recommend in general, except for deeply coupled co-developed packages where unstable APIs are used.

**Comment by trungleduc** [comment]
I agree with @minrk, bounding the version from 0.4 to 0.11 seems redundant. What do you think @martinRenou @jtpio

**Comment by jtpio** [comment]
Sounds good. And CI can help catch potential issues with future versions.

**Review by trungleduc — APPROVED** [review]

**Pull request merged**
